configure.ac

```diff
@@ -899,14 +923,15 @@ done
 if test "$HAVE_CXXSTD" = TRUE; then
     AC_MSG_RESULT([yes ($CXXFLAGS_CXXSTD)])
 else
-    AC_MSG_ERROR(no)
+    AC_MSG_RESULT([no])
+    AC_MSG_ERROR(Compiler doesn't support C++17)
 fi
 
 AS_IF([test "$ENABLE_GTKAPP" != true],
 [CXXFLAGS="$CXXFLAGS $CXXFLAGS_CXXSTD"])
 
 STD_FILESYSTEM=
-if test "$CXXFLAGS_CXXSTD" = "-std=c++17" -a "$ENABLE_ANDROIDAPP" != "true" ; then
+if test "$HAVE_CXXSTD" = TRUE -a "$ENABLE_ANDROIDAPP" != "true" ; then
```

Comment:
Could also be `-std=c++1z`, but since we have `$HAVE_CXXSTD` if C++17 is available, best to check that instead.